### PR TITLE
Inference updates to add locust swarming

### PR DESCRIPTION
**Pull request by gushob21**

### CODE REVIEW

File: use-cases/inferencing/benchmark/src/locust-tasks/logging.conf [hunk 1]
```diff
@@ -0,0 +1,24 @@
+[formatters]
+keys=custom_json
+
+[formatter_custom_json]
+class=custom_json_formatter.CustomJSONFormatter
```
Comment (arueth) [inline use-cases/inferencing/benchmark/src/locust-tasks/logging.conf:5]: You need to bring this file into your src directory

https://github.com/GoogleCloudPlatform/accelerated-platforms/blob/fix-locust-benchmarking/use-cases/model-fine-tuning-pipeline/data-processing/ray/src/custom_json_formatter.py

File: use-cases/inferencing/benchmark/src/locust-tasks/requirements.txt [hunk 2]
```diff
@@ -1,31 +1,2 @@
...
-requests==2.27.1
-roundrobin==0.0.2
-six==1.16.0
-typing_extensions==4.1.1
-urllib3==1.26.9
-Werkzeug==2.1.1
-zipp==3.8.0
-zope.event==4.5.0
-zope.interface==5.4.0
+requests==2.27.1
```
Comment (arueth) [inline use-cases/inferencing/benchmark/src/locust-tasks/requirements.txt:2]: Can you add a newline at the end

File: use-cases/inferencing/benchmark/src/locust-tasks/tasks.py [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
+#!/usr/bin/env python
+
+# Copyright 2022 Google Inc. All rights reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
 #
-#      http://www.apache.org/licenses/LICENSE-2.0
+#     http://www.apache.org/licenses/LICENSE-2.0
```
Comment (arueth) [inline use-cases/inferencing/benchmark/src/locust-tasks/tasks.py:9]: I'm not sure why this changed.😀

File: use-cases/inferencing/benchmark/src/locust-tasks/logging.conf [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+formatter=custom_json
+
+[loggers]
+keys=root
+
+[logger_root]
+level=INFO
+handlers=console
+
+[logger_model_eval]
```
Comment (arueth) [inline use-cases/inferencing/benchmark/src/locust-tasks/logging.conf:22]: Your logger has to match your getLogger value

logger = logging.getLogger("benchmark_obj")

**Review by arueth — CHANGES_REQUESTED** [review]

**Comment by gushob21** [comment]
Hey @arueth , made changes as you suggested,. Additionally, I got rid of the dir `locst-task` and moved the scripts in there directly under src folder to keep it aligned with other guides.

File: use-cases/inferencing/benchmark/src/Dockerfile [hunk 5]
```diff
@@ -16,20 +16,18 @@
...
+RUN pip3 install --no-cache-dir -r requirements.txt
+
+COPY custom_json_formatter.py logging.conf run.sh tasks.py /
 
 # Expose the required Locust ports
 EXPOSE 5557 5558 8089
 
 # Set script to be executable
-RUN chmod 755 /locust-tasks/run.sh
+RUN chmod 755 ./run.sh
```
Comment (arueth) [inline use-cases/inferencing/benchmark/src/Dockerfile:30]: Make this an absolute path `/run.sh`

File: use-cases/inferencing/benchmark/src/Dockerfile [hunk 6]
```diff
@@ -16,20 +16,18 @@
...
 
 # Start Locust using LOCUS_OPTS environment variable
-ENTRYPOINT ["/locust-tasks/run.sh"]
+ENTRYPOINT ["./run.sh"]
```
Comment (arueth) [inline use-cases/inferencing/benchmark/src/Dockerfile:33]: Make this an absolute path `/run.sh`

File: use-cases/inferencing/benchmark/src/Dockerfile [hunk 7]
```diff
@@ -16,20 +16,18 @@
 # Start with a base image Python 3.9.12 Debian 11 (bullseye) slim 
 FROM python:3.9.12-slim-bullseye
```
Comment (arueth) [inline use-cases/inferencing/benchmark/src/Dockerfile:17]: Is there a requirement to use 3.9.12 and Debian? Could we update to 3.13.0-alpine if possible?

File: use-cases/inferencing/benchmark/src/requirements.txt [hunk 8]
```diff
@@ -0,0 +1,2 @@
+locust==2.8.6
```
Comment (arueth) [inline use-cases/inferencing/benchmark/src/requirements.txt:1]: Is there any requirement to be on 2.7.6? Could we go to the latest 2.32.2?

File: use-cases/inferencing/benchmark/src/requirements.txt [hunk 9]
```diff
@@ -0,0 +1,2 @@
...
+requests==2.27.1
```
Comment (arueth) [inline use-cases/inferencing/benchmark/src/requirements.txt:2]: Is there a requirement to be on 2.27.1? Could we go to the latest 2.32.3?

**Review by arueth — CHANGES_REQUESTED** [review]

**Review by arueth — APPROVED** [review]

**Pull request merged**